detekt-generator/src/main/kotlin/io/gitlab/arturbosch/detekt/generator/collection/DocumentationCollector.kt

```diff
@@ -0,0 +1,83 @@
+package io.gitlab.arturbosch.detekt.generator.collection
+
+import io.gitlab.arturbosch.detekt.generator.collection.exception.InvalidCodeExampleDocumentationException
+import org.jetbrains.kotlin.psi.KtClassOrObject
+
+class DocumentationCollector {
+
+    private var name: String = ""
+    var description: String = ""
+        private set
+    var compliant: String = ""
+        private set
+    var nonCompliant: String = ""
+        private set
+
+    fun setClass(classOrObject: KtClassOrObject) {
+        name = classOrObject.name?.trim() ?: ""
+        classOrObject.kDocSection()
+            ?.getContent()
+            ?.trim()
+            ?.replace("@@", "@")
+            ?.also { comment ->
+                extractRuleDocumentation(comment)
+            }
```

Comment:
```suggestion
            ?.let(::extractRuleDocumentation)
```
`also` is confusing because we are not preserving the evaluated value of the previous expression.